### PR TITLE
feat: add C implementation for `math/base/special/round`

### PR DESCRIPTION
**Pull request by Pranavchiku**
## Description

> What is the purpose of this pull request?

This PR adds C implementation for `@stdlib/math/base/special/round`

## Related Issues

> Does this pull request have any related issues?

## Questions

> Any questions for reviewers of this pull request?

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Pranavchiku** [comment]
cc @kgryte @Planeshifter

**Comment by kgryte** [comment]
@Pranavchiku Note that you previously worked on a PR for this feature: https://github.com/stdlib-js/stdlib/pull/745

**Comment by Pranavchiku** [comment]
Reference implementation: https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-math.round

Not sure how much credible this source is, but aligns with other miscellaneous sources.

**Comment by kgryte** [comment]
@Pranavchiku That source is credible. That is the official ECMAScript specification from TC39.

**Comment by Pranavchiku** [comment]
I think this is ready to review.

File: lib/node_modules/@stdlib/math/base/special/round/src/main.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include "stdlib/math/base/special/round.h"
+#include "stdlib/math/base/special/floor.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/assert/is_negative_zero.h"
+
+/**
+* Rounds a numeric value to the nearest integer.
+*
+* @param x    input value
+* @return	  output value
```
Comment (USERSATOSHI) [inline lib/node_modules/@stdlib/math/base/special/round/src/main.c]: can be style fix by removing extra space 
```c
* @param x    input value
* @return	  output value
```
to
```c
* @param x    input value
* @return     output value
```

Comment (USERSATOSHI) [reply lib/node_modules/@stdlib/math/base/special/round/src/main.c]: But rest of it LGTM

Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/round/src/main.c:28]:
```suggestion
* @return.     output value
```

Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/round/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

**Review by Planeshifter — APPROVED** [review]
Looks good to me, so let's get this in as the C implementation is required for  the one of `rempio2`, which @gunjjoshi will be working on soon.

**Pull request merged**